crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -94,7 +97,7 @@ impl NamedChannelOrUrl {
                 }
                 base_url
```

Comment:
This is not guaranteed to have a trailing slash now.

Reply:
Maybe you can also add a comment to the function that the returned url will always have a trailing slash?

Reply:
And also add some tests to verify that base_url does always have a trailing slash but as_str doesnt.

Reply:
I think i've added what you requested in this comment but I'm not sure what you mean with the `as_str`. The canonical name is tested in:
https://github.com/mamba-org/rattler/blob/68c4254de11293639a5800b6f05295405b2d5eff/crates/rattler_conda_types/src/channel/mod.rs#L654-L673